artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerSessionImpl.java

```diff
@@ -175,7 +176,7 @@
    private final OperationContext context;
 
    // Session's usage should be by definition single threaded, hence it's not needed to use a concurrentHashMap here
-   protected final Map<SimpleString, Pair<Object, AtomicLong>> targetAddressInfos = new HashMap<>();
+   protected final Map<SimpleString, Pair<Object, AtomicLong>> targetAddressInfos = new MaxSizeMap<>(100);
```

Comment:
Can this be configurable

Reply:
Yes, it could be configurable and you're right that 100 is arbitrary but so would any other not-ulimited value.

Reply:
Will this represent any error in case the size > 100 or this is just for information?

If I send to a destination that was cleaned before, will I be able to send it again?


I believe this is just for informational purposes on the Console?

Reply:
There won't be any error. The map will just silently drop the oldest entry when the maxSize is hit. You'll be able to send to any destination again that was dropped previously due to maxSize. This information is only for administrative use.

Reply:
@michaelandrepearce / @jbertram I don't know.. we already have too many options to configure. I don't foresee anyone actually needing to tweak this value. Can't we just leave it be?

You are already removing temporary destinations from the session. And this is for the corner case where the Session sending the temporary destination is a different session.